cmd/bucket-quota.go

```diff
@@ -60,23 +60,50 @@ func (sys *BucketQuotaSys) Remove(bucketName string) {
 	sys.Unlock()
 }
 
-// Exists - bucketName has Quota config set
-func (sys *BucketQuotaSys) Exists(bucketName string) bool {
-	sys.RLock()
-	_, ok := sys.quotaMap[bucketName]
-	sys.RUnlock()
-	return ok
-}
-
-// Keys - list of buckets with quota configuration
-func (sys *BucketQuotaSys) Keys() []string {
+// Buckets - list of buckets with quota configuration
+func (sys *BucketQuotaSys) Buckets() []string {
 	sys.RLock()
 	defer sys.RUnlock()
-	var keys []string
+	var buckets []string
 	for k := range sys.quotaMap {
-		keys = append(keys, k)
+		buckets = append(buckets, k)
+	}
+	return buckets
+}
+
+// Init initialize bucket quota sys configuration with all buckets.
+func (sys *BucketQuotaSys) Init(buckets []BucketInfo, objAPI ObjectLayer) error {
+	if objAPI == nil {
+		return errServerNotInitialized
+	}
+
+	// In gateway mode, we always fetch the bucket lifecycle configuration from the gateway backend.
```

Comment:
// In gateway mode, we always fetch the bucket quota configuration from the gateway backend.